documentation/docs/extensibility.md

```diff
@@ -1,86 +1,199 @@
 # Extensibility
 
-There are several possibilities for extensibility besides the **[very powerful configuration](configuration.md)**:
+When using one of the ready-made pipelines project "Piper" provides, the basic idea is to not write custom pipeline code.
+The pipelines are centrally maintained, and can be used with a small amount of [configuration](configuration.md).
 
-## 1. Stage Exits
+For the large majority of _standard_ projects, the features of the ready-made pipelines should be enough to implement a production-ready CI/CD workflow with little effort.
 
-  You have to create a file like `<StageName>.groovy` (for example, `Acceptance.groovy`) and store it in folder `.pipeline/extensions/` in your source code repository.
+This approach allows you to focus on what you need to get done while implementing [Continuous Delivery](https://martinfowler.com/bliki/ContinuousDelivery.html) in a best-practice compliant way.
 
-!!! note "Cloud SDK Pipeline"
-    If you use the Cloud SDK Pipeline, the folder is named `pipeline/extensions/` (without the dot). For more information, please refer to [the Cloud SDK Pipeline documentation](https://github.com/SAP/cloud-s4-sdk-pipeline/blob/master/doc/pipeline/extensibility.md).
+If a feature you need is missing, or you discovered a bug in one of the ready-made pipelines, please see if there is already an [issue in our GitHub repository](https://github.com/SAP/jenkins-library/issues), and open a new one if that is not the case.
 
-  The pipeline template checks if such a file exists and executes it, if present.
-  A parameter that contains the following keys is passed to the extension:
+In some cases, specialised features might not be desirable for inclusion in the ready-made pipelines.
+You can still benefit from the qualities they provide if you can address your requirements via an **Extension**.
 
-  * `script`: defines the global script environment of the Jenkinsfile run. This makes sure that the correct configuration environment can be passed to project "Piper" steps and also allows access to for example the `commonPipelineEnvironment`.
-  * `originalStage`: this will allow you to execute the "original" stage at any place in your script. If omitting a call to `originalStage()` only your code will be executed instead.
-  * `stageName`: name of the current stage
-  * `config`: configuration of the stage (including all defaults)
+Extensions are custom bits of pipeline coding that you can use to implement special requirements.
 
-  Here a simple example for such an extension:
+Before building extensions, please make sure that no better alternative works for you.
```

Comment:
```suggestion
Before building extensions, please make sure that there really is no better alternative.
```
or
```suggestion
Before building extensions, please make sure that there is no alternative which works better for you.
```

"no better alternative works for you" implies that there can be a better alternative, but it doesn't work for you.